server.go

```diff
@@ -71,28 +73,53 @@ type Server struct {
 	OnError func(w http.ResponseWriter, r *http.Request, e error)
 }
 
+type ServerOption = func(*Server)
+
 type routeContextKey struct{}
 type parametersContextKey struct{}
 
 const defaultTimeout = 10 * time.Second
 
 // NewServer returns a new Server that will make requests to the given target argument.
-func NewServer(target string) *Server {
-	return &Server{
+func NewServer(target string, opts ...ServerOption) *Server {
+	server := &Server{
 		MultiplexerTripper: multiplexer.NewStandardTripper(&http.Client{}),
 		Logger:             log.Default(),
 		SecretFilter:       secretfilter.New(),
 		Addr:               "localhost:3005",
 		ProxyTimeout:       defaultTimeout,
 		ReadTimeout:        defaultTimeout,
 		WriteTimeout:       defaultTimeout,
-		PassThrough:        false,
+		passThrough:        false,
 		AroundRequest:      func(h http.Handler) http.Handler { return h },
 		target:             target,
 		ignoreHeaders:      make(map[string]bool),
 		routes:             make([]Route, 0),
 		tracingConfig:      tracing.TracingConfig{Enabled: false},
 	}
+
+	for _, fn := range opts {
+		fn(server)
+	}
+
+	return server
+}
+
+func WithPassThrough(passthroughTarget string) ServerOption {
+	targetURL, err := url.Parse(passthroughTarget)
+
+	if err != nil {
+		panic(err)
```

Comment:
I think it could be worth returning this error through back to callers of `NewServer` instead of panicing.

So changing `ServerOption` to return an `error` and `NewServer` to return an `error`

Reply:
I don't think most upstreams would be able to recover from a misconfigured URL (e.g. we'll probably log and panic), but it's probably better to give them the option vs crashing immediately.

I'll update this 👍